eventmesh-connectors/eventmesh-connector-http/src/main/java/org/apache/eventmesh/connector/http/common/BoundedConcurrentQueue.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.eventmesh.connector.http.common;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * <p>BoundedConcurrentQueue is a wrapper class for ConcurrentLinkedQueue</p>
+ * <ol>
+ * <li>Limit the maximum size of the queue</li>
+ * <li>Add an object to the queue, if the queue is full, remove the head element and add the new element.(thread safe)</li>
```

Comment:
Ok, I'll look into those queues and see if that's a better solution

Reply:
Both of these queues can fulfill the current demand, but they are not thread-safe.

Reply:
How about synchronize them when calling these queues? You can do this in the logic code, or in a custom class.

Reply:
You can also pick a Deque implementation.

Reply:
The sink and source in the previous code both involve the same queue operation, so I extracted and encapsulated them to reduce duplication of code. Is there anything wrong with the encapsulation of this class? Based on my limited knowledge, it should be able to meet the current needs.

Reply:
Since you had some questions about the way this BundledQueue was written and synchronized, I've suggested some similar templates in the spirit of “not reinventing the wheel”. It's up to you to evaluate what is the cleanest and most maintainable way to do this at the code level, whether it's a simple wrapper around a Queue in a third-party utility, or continuing to use the BundledQueue.

Reply:
Ok, I'll evaluate the class again in terms of business logic to see if I have to do this, I'll give my answer in a few days

Reply:
Conversation resolved by https://github.com/apache/eventmesh/pull/4913#issuecomment-2191773641.